app/models/criterion.rb

```diff
@@ -140,6 +143,7 @@ def self.update_assigned_groups_counts(assignment)
 
   # When max_mark of criterion is changed, all associated marks should have their mark value scaled to the change.
   def scale_marks
+    self.assignment.update_results_stats if previous_changes.key?('bonus')
```

Comment:
Hmm interesting. I'm thinking that maybe `update_results_stats` should be called when the `max_mark` is changed as well, but called after the marks are updated (e.g. line 171).

Reply:
In that case I would essentially change the guard clause to an `if` branch right? Because, I am essentially piggybacking off a callback that only serves the purpose of scaling marks and am now using the callback to update results based on a bonus designation.

Reply:
Yes that's true, I'm a bit torn. Another approach would be to make a new callback function (e.g., "update result stats") that you have in an `after_update`, and then do

```ruby
def update_result_stats
  return unless max_mark changed or bonus changed
  scale marks if max_mark_changed
  self.assignment.update_results_stats
```

Reply:
Hm ok, I have implemented the approach I mentioned, but I can easily change the approach to this one as well.